internal/controller/power_monitor_internal.go

```diff
@@ -53,28 +61,67 @@ type PowerMonitorInternalReconciler struct {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *PowerMonitorInternalReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Set up indexer for PowerMonitorInternal based on referenced ConfigMaps
+	err := mgr.GetFieldIndexer().IndexField(context.TODO(), &v1alpha1.PowerMonitorInternal{}, configMapField, func(obj client.Object) []string {
+		pmi := obj.(*v1alpha1.PowerMonitorInternal)
```

Comment:
Lets check if the conversion is okay and exit if it fails. (May be log it as well)